Sources/Workspace/Workspace.swift

```diff
@@ -2763,214 +2923,84 @@ extension Workspace {
-                    // Impossible
-                    break
-                case .revision(let revision):
-                    result.append("resolved to '\(revision)'")
-                case .unversioned:
-                    result.append("unversioned")
-                }
-            case .edited?:
-                result.append("edited")
-            case .local?:
-                result.append("versioned")
-            case nil:
-                result.append("root")
-            }
-
-            result.append(" but now has a ")
-
-            switch requirement {
-            case .versionSet:
-                result.append("different version-based")
-            case .revision:
-                result.append("different revision-based")
-            case .unversioned:
-                result.append("unversioned")
-            }
-
-            result.append(" requirement.")
-        default:
-            result.append(" requirements have changed.")
-        }
-
-        return result
-    }
-}
-
-/// A result which can be loaded.
-///
-/// It is useful for objects that holds a state on disk and needs to be
-/// loaded frequently.
-public final class LoadableResult<Value> {
-    /// The constructor closure for the value.
-    private let construct: () throws -> Value
-
-    /// Create a loadable result.
-    public init(_ construct: @escaping () throws -> Value) {
-        self.construct = construct
-    }
-
-    /// Load and return the result.
-    public func loadResult() -> Result<Value, Error> {
-        return Result(catching: {
-            try self.construct()
-        })
-    }
-
-    /// Load and return the value.
-    public func load() throws -> Value {
-        return try loadResult().get()
+        try self.repositoryManager.remove(repository: dependency.packageRef.makeRepositorySpecifier())
     }
 }
 
-private struct RemoteArtifact {
-    let packageRef: PackageReference
-    let targetName: String
-    let url: Foundation.URL
-    let checksum: String
-}
-
-private struct ArchiveIndexFile: Decodable {
-    let schemaVersion: String
-    let archives: [Archive]
 
-    struct Archive: Decodable {
-        let fileName: String
-        let checksum: String
-        let supportedTriples: [Triple]
+// MARK: - Utility extensions
 
-        enum CodingKeys: String, CodingKey {
-            case fileName
-            case checksum
-            case supportedTriples
-        }
-
-        public init(from decoder: Decoder) throws {
-            let container = try decoder.container(keyedBy: CodingKeys.self)
-            self.fileName = try container.decode(String.self, forKey: .fileName)
-            self.checksum = try container.decode(String.self, forKey: .checksum)
-            self.supportedTriples = try container.decode([String].self, forKey: .supportedTriples).map(Triple.init)
-        }
-    }
-}
-
-private extension Workspace.ManagedArtifact {
+fileprivate extension Workspace.ManagedArtifact {
```

Comment:
the diff above is a mess, but nothing material has changed, just moved methods around to better organize the code in this huge file